email_handler.py

```diff
@@ -465,6 +468,23 @@ def handle_reply(envelope, smtp: SMTP, msg: Message, rcpt_to: str) -> (bool, str
         handle_bounce(contact, alias, msg, user, mailbox_email)
         return False, "550 SL E6"
 
+    mailb: Mailbox = Mailbox.get_by(email=mailbox_email)
+    if ENFORCE_SPF_AVAILABLE and mailb.force_spf:
+        if msg['X-SimpleLogin-Client-IP']:
+            r = spf.check2(i=msg['X-SimpleLogin-Client-IP'], s=envelope.mail_from.lower(), h=None)
+            # TODO: Handle temperr case (e.g. dns timeout)
+            # only an absolute pass, or no SPF policy at all is 'valid'
+            if r[0] not in ["pass", "none"]:
+                LOG.d(
+                    "SPF validation failed for %s (reason %s)", mailbox_email, r[0],
+                )
+                # TODO: Send warning to mailbox
```

Comment:
In case of impersonation attacks, we can send **too many** alert emails to user. I think it's relatively safe to just return an error here as if the email is sent from the user's mailbox, it should receive an error anyway.